components/ILIAS/Mail/classes/class.ilMailFormGUI.php

```diff
@@ -175,20 +177,126 @@ protected function decodeAttachmentFiles(array $files): array
         return $decoded_files;
     }
 
-    public function sendMessage(): void
+    public function saveMessageToOutbox(): void
     {
-        $message = $this->getBodyParam('m_message', $this->refinery->kindlyTo()->string(), '');
+        $schedule_datetime = new DateTimeImmutable(
+            $this->getBodyParam('m_schedule', $this->refinery->kindlyTo()->string(), null),
+            new DateTimeZone($this->user->getTimezone())
+        );
+
+        $files = $this->decodeAttachmentFiles($this->getBodyParam(
+            'attachments',
+            $this->refinery->kindlyTo()->listOf(
+                $this->refinery->custom()->transformation($this->refinery->kindlyTo()->string())
+            ),
+            []
+        ));
 
-        $mail_body = new ilMailBody($message, $this->purifier);
 
+        $rcp_to = ilUtil::securePlainString($this->getBodyParam('rcp_to', $this->refinery->kindlyTo()->string(), ''));
+        $rcp_cc = ilUtil::securePlainString($this->getBodyParam('rcp_cc', $this->refinery->kindlyTo()->string(), ''));
+        $rcp_bcc = ilUtil::securePlainString($this->getBodyParam('rcp_bcc', $this->refinery->kindlyTo()->string(), ''));
+
+        $message = ilUtil::securePlainString($this->getBodyParam('m_message', $this->refinery->kindlyTo()->string(), ''));
+        $mail_body = new ilMailBody($message, $this->purifier);
         $sanitized_message = $mail_body->getContent();
 
+        $outbox_folder_id = $this->mbox->getOutboxFolder();
+        if (ilSession::get('outbox')) {
+            $outbox_id = (int) ilSession::get('outbox');
+            ilSession::clear('outbox');
+        }
+
+        $this->umail->sendScheduledMail(
+            $outbox_id ?? 0,
+            $outbox_folder_id,
+            $this->user->getId(),
+            $files,
+            $rcp_to,
+            $rcp_cc,
+            $rcp_bcc,
+            ilUtil::securePlainString(
+                $this->getBodyParam('m_subject', $this->refinery->kindlyTo()->string(), '')
+            ) ?: 'No Subject',
+            $sanitized_message,
+            $schedule_datetime,
+            new DateTimeZone($this->user->getTimeZone()),
```

Comment:
The timezone can be IMO retrieved from `$schedule_datetime`. As written above, I doubt that we need this at all.